docs/tutorials/custom/custom-base-dog.md

```diff
@@ -345,42 +345,58 @@ Name your module `my-custom-base`.
-Navigate to the **Components** subtab of the **Config** tab.
+Navigate to the **Components** subtab of your robot's **Config** tab.
 
-### Configure the base
+### Configure the camera
 
-In the **Create Component** field, give your base a name.
-We called ours "quadruped".
-In the **Type** drop-down select `base`.
-In the **Model** field, type in `viamlabs:base:robotdog`.
-Click **Create Component**.
+Configure the ribbon camera on the dog as a `webcam` following our [webcam documentation](/components/camera/webcam/).
 
-![Screenshot of the Viam app CONFIG tab with the Config subtab open, showing quadruped configured.](/tutorials/custom-base-dog/config-modular-component.png)
+Click **Save config**.
 
-You need to add the `ip_address` and `port` attributes to your base config.
-In the attributes field, paste the following, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`):
+### Configure the base
+
+Because your custom base relies on a local module, you need to use raw JSON to configure your modular resource.
+Use the **Mode** selector in the upper-left corner of the **Config** tab to switch to **Raw JSON** mode.
+
+Locate the `"components": []` portion of the config file.
+At the end of your camera configuration, add a comma and then add the following base configuration:
 
 ```json
-{
-  "ip_address": "<HOSTNAME>.local",
-  "port": 5001
-}
+    {
+      "namespace": "rdk",
+      "name": "quadruped",
+      "type": "base",
+      "model": "viamlabs:base:robotdog",
+      "attributes": {
+          "ip_address": "<HOSTNAME>.local",
+        "port": 5001
+      },
+      "depends_on": []
+    }
 ```
 
+Edit the `ip_address` attribute to match your robot's IP address, replacing `<HOSTNAME>` with your Pi's hostname (for example, `"ip_address": "robotdog.local"`).
 If this doesn't work, you can instead try using the IP address of the machine where the module is running, for example, `"ip_address": "10.0.0.123"`.
 
+If you are using a port other than `5001`, edit the `port` attribute.
 `5001` is the default port for sending and receiving instructions to and from the Freenove server.
 
-Click **Save Config**.
+Click **Save config**.
 
-### Configure the camera
+Your raw JSON configuration should look similar to the following:
 
-Configure the ribbon camera on the dog as a `webcam` following our [webcam documentation](/components/camera/webcam/).
+![Raw JSON mode on the Config tab, showing the components and modules sections of the robot's raw JSON config.](/tutorials/custom-base-dog/raw-json.png)
+
+Toggle back to **Builder** mode and make sure a card has been generated:
+
+![Screenshot of the Viam app CONFIG tab with the Config subtab open, showing quadruped configured.](/tutorials/custom-base-dog/config-modular-component.png)
+
+If yours doesn't look like this, go back to the raw JSON and double-check your JSON formatting.

```

Comment:
```suggestion
If your configuration does not match, go back to the raw JSON and double-check your JSON formatting.

```

Reply:
```suggestion
If you don't have a correctly-generated configuration panel, go back to the raw JSON and double-check your JSON formatting.
```
I don't want to say "match" since they may have slightly different attribute values. How is this?

Reply:
Ah, this hints that we might fail to generate the config correctly on our end (like: "unplug it and plug it back in again" style), while the issue is with config entry on the part of the user (hence: "double check your work"). You're right about `match`; I guess I'm just trying not to use "yours doesn't look like this". Since the screenshot may well differ from the actual config, what about language like the following instead, but introducing the screenshot instead of following it:

> Your configuration should resemble the following:

Reply:
```suggestion
If yours doesn't resemble the following, go back to the raw JSON and double-check your JSON formatting.
```
Maybe this? I don't want to lose the main point of this which is to get them to make sure their JSON punctuation and indentation is correct, since we usually guide users to fancy config but can't here.

Reply:
Gotcha, works for me -- thank you!